actix-tracing/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+//! Actix tracing - support for tokio tracing with Actix services.
+#![deny(rust_2018_idioms, warnings)]
+
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+
+use actix_service::{apply, IntoServiceFactory, Service, ServiceFactory, Transform};
+use futures_util::future::{ok, Either, Ready};
+use tracing_futures::{Instrument, Instrumented};
+
+/// Trait representing a type that is able to produce spans for a given value.
+///
+/// The generic argument `T` is typically the request type of a `Service`
+/// implementation.
+pub trait MakeSpan<T> {
```

Comment:
what is the reason for trait here? why not just simple `Fn(&T) -> Option<Span>`?

Reply:
Made it a trait in the anticipation that we might have more span making functions in the future (for `poll_ready` for instance) and provided a blanket impl for `Fn(&T) -> Option<Span>`. But I don't feel strongly about this. We can always have a trait later if we think its appropriate. Will change.